src/System.IO.UnmanagedMemoryStream/tests/Uma.ReadWriteStructArray.cs

```diff
@@ -0,0 +1,179 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public class Uma_ReadWriteStructArray
+    {
+        private const int UmaTestStruct_DisalignedSize = 2 * sizeof(int) + 2 * sizeof(bool) + sizeof(char);
+        private const int UmaTestStruct_AlignedSize = 16; // potentially architecture dependent.
+        private struct UmaTestStruct
+        {
+            public int int1;
+            public bool bool1;
+            public int int2;
+            public char char1;
+            public bool bool2;
+        }
+
+        private struct UmaTestStruct_ContainsReferenceType
+        {
+            public object referenceType;
+        }
```

Comment:
Could we put these helper types in a shared file or a base class or somewhere so we don't have to redefine them in multiple places?